Updated Impeller details (#8607)
flutter/website: src/perf/impeller.md

---
title: Impeller rendering engine
description: What is Impeller and how to enable it?
---

## What is Impeller?

Impeller provides a new rendering runtime for Flutter.

> The Flutter team's believes this solves Flutter's

Reviewer: ```suggestion
The Flutter team believes this solves Flutter's
```

[What is Impeller?](https://www.youtube.com/watch?v=gKrYWC_SDxQ)

Impeller has the following objectives:

* **Predictable performance**:
  Impeller compiles all shaders and reflection offline at build time.
  It builds all pipeline state objects upfront.
  The engine controls caching and caches explicitly.
* **Instrumentable**:
  Impeller tags and labels all graphics resources like textures,
  and buffers.
  It can capture and persist animations to disk without affecting
  per-frame rendering performance.
* **Portable**:
  Flutter doesn't tie Impeller to a specific client rendering API.
  You can author shaders once and convert them to backend-specific
  formats as necessary.
* **Leverages modern graphics APIs**:
  Impeller uses, but doesn’t depend on, features available in
  modern APIs like Metal and Vulkan.
* **Leverages concurrency**:
  Impeller can distribute single-frame workloads across multiple
  threads if necessary.

## Availability

Where can you use Impeller?

### iOS

Flutter enables Impeller by default on iOS.

> To _disable_ Impeller on iOS, take one of the following actions:

Reviewer 2: Worth noting that --disable-impeller is only for debugging and if you want to _ship_ with impeller off you must add it to the Plist entry.

  pass `--disable-impeller` to the `flutter run` command.

  ```terminal
  $ flutter run --disable-impeller
  ```

* To _disable_ Impeller on iOS when deploying your app,
  add the following tags under the top-level `<dict>` tag in your
  app's `Info.plist` file.

  ```xml
    <key>FLTDisableImpeller</key>
    <true/>
  ```

The team continues to improve iOS support.
If you encounter performance or fidelity issues
with Impeller on iOS, file an issue in the [GitHub tracker][file-issue].
Prefix the issue title with `[Impeller]` and
include a small reproducible test case.

[file-issue]: {{site.github}}/flutter/flutter/issues/new/choose

### Android

Android development continues but it's not ready for preview.

> On the `master` channel, it might not work at the tip-of-tree.

Reviewer: It looks like this sentence got garbled. Maybe consider:

```suggestion
Impeller on Android might not work on the `master` channel.
```

To see what direction Android support will take,
experiment with Impeller in the 3.7 or later stable release.

To enable Impeller on Android, take one of the following actions:

* Pass `--enable-impeller` to the `flutter run` command.

  ```terminal
  $ flutter run --enable-impeller
  ```

* Add the following tag to your `AndroidManifest.xml` file
  under the `<application>` tag.

  ```xml
    <meta-data
      android:name="io.flutter.embedding.android.EnableImpeller"
      android:value="true" />
  ```

## Architecture

To learn more details about Impeller's design and architecture,
check out the [README.md][] file in the source tree.

[README.md]: {{site.github}}/flutter/engine/blob/main/impeller/README.md

> ## Additional Information

Reviewer 3: Should use sentence case.
